### PR TITLE
Add SonarCloud to GitHub Actions

### PR DESCRIPTION
**Pull request by dhaiducek**
Addresses:
- https://github.com/stolostron/backlog/issues/10735

Dependency on:
- https://github.com/openshift/release/pull/25900

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -109,14 +112,23 @@ lint: lint-dependencies lint-all
...
+
+test-coverage: TESTARGS = -json -coverprofile=coverage_unit.out
+test-coverage: test
 
 test-dependencies:
 	curl -L https://github.com/kubernetes-sigs/kubebuilder/releases/download/v$(KBVERSION)/kubebuilder_$(KBVERSION)_$(GOOS)_$(GOARCH).tar.gz | tar -xz -C /tmp/
 	sudo mv /tmp/kubebuilder_$(KBVERSION)_$(GOOS)_$(GOARCH) /usr/local/kubebuilder
 
+gosec-install:
+	curl -sfL https://raw.githubusercontent.com/securego/gosec/master/install.sh | sh -s -- -b $(shell pwd)/bin
```
Comment (mprahl) [inline Makefile:128]: Do you think you could use `go get` to compile it or at the very least target a specific tag/commit? It seems a little dangerous to be running a script right from the master branch in the event the repo gets compromised.

File: .github/workflows/kind.yml [hunk 2]
```diff
@@ -81,12 +81,22 @@ jobs:
...
+    - name: Test Coverage and Report Generation
+      if: ${{ matrix.kind == 'latest' }}
       run: |
-        echo "SonarCloud not implemented."
+        make test-coverage | tee report_unit.json
+        make gosec-install
+        make gosec-scan
+        cat gosec.json
+
+    # - name: SonarCloud Analysis
```
Comment (mprahl) [inline .github/workflows/kind.yml:94]: Why is this commented out?

Comment (dhaiducek) [reply .github/workflows/kind.yml:94]: I didn't want my tests to duplicate the scans on Prow while they were still enabled. I'll remove this commit once they're gone in Prow.

File: Makefile (hunk 1 above)
Comment (dhaiducek) [reply Makefile:128]: Yes, I can do that!

Comment (dhaiducek) [reply Makefile:128]: Although, this is consistent with what CICD uses in Prow, so if the repo is compromised we're all in trouble...

Comment (mprahl) [reply Makefile:128]: @dhaiducek but we can always strive to do better.  :)

Comment (dhaiducek) [reply Makefile]: Indeed we can 🙂 Updated!

I tried the `go get` method, but it doesn't spit out a version when running `gosec --version` that way, so I think the tarball may be the way to go.

**Review by mprahl — DISMISSED** [review]

**Comment by dhaiducek** [comment]
Shoot. This isn't going to work because secrets aren't available to forks. Discussion thread started in Slack: https://coreos.slack.com/archives/GUQS3CS5A/p1643841820681019

**Comment by dhaiducek** [comment]
FYI: It's not actually ready for review--I just wanted to kick off the Prow jobs. Feel free to Ignore/Unsubscribe/Unassign yourself if you want to not be bothered by all the noise I'm creating. 🙂

**Comment by dhaiducek** [comment]
Okay, this PR is ready for reviews.

**Review by ycao56 — APPROVED** [review]

**Pull request merged**